modules/transaction-payment/src/lib.rs

```diff
@@ -1359,6 +1358,10 @@ where
 				refund = refund_fee.saturating_add(refund_tip);
 				actual_tip = tip.saturating_sub(refund_tip);
 			}
+			// fee surplus not included in refund
+			if let Some(surplus) = surplus {
+				refund = refund.saturating_sub(surplus);
```

Comment:
we still want to do a partial refund or surplus similar how we refund tip

Reply:
actual_surplus=actual_fee * percent, refund_surplus=surplus-actual_surplus, should also passing `percent` as `Pre`.